packages/search/src/features/registration/deduplicate/service.ts

```diff
@@ -140,14 +147,71 @@ export const searchForBirthDuplicates = async (
             boost: 1
           }
         }
-      ].filter(Boolean)
+      ].filter(isNonEmptyCondition)
+    }
+  }
+
+  const childsNameMatch = {
+    bool: {
+      must: {
+        bool: {
+          must: [
```

Comment:
```suggestion
    bool: {
      must: [
```
We probably don't the extra nesting right? It wasn't there previously too